crates/vhost/src/backend.rs

```diff
@@ -72,6 +75,71 @@ pub struct VhostUserMemoryRegionInfo {
     pub mmap_offset: u64,
     /// Optional file descriptor for mmap.
     pub mmap_handle: RawFd,
+
+    #[cfg(feature = "xen")]
+    /// Xen specific flags.
+    pub xen_mmap_flags: u32,
+
+    #[cfg(feature = "xen")]
+    /// Xen specific data.
+    pub xen_mmap_data: u32,
+}
+
+impl VhostUserMemoryRegionInfo {
+    /// Creates Self from GuestRegionMmap.
+    pub fn from_guest_region<B: Bitmap>(region: &GuestRegionMmap<B>) -> Result<Self> {
+        let (mmap_handle, mmap_offset) = match region.file_offset() {
+            Some(file_offset) => (file_offset.file().as_raw_fd(), file_offset.start()),
+            None => return Err(Error::InvalidGuestMemoryRegion),
+        };
```

Comment:
probably would have used `.ok_or[_else](...)?` here. But I do not think this code is complex either...